fix(sidebar): open menu, highlight current page

caugner opened the pull request:
## Summary

Highlights the current page in the sidebar, and expands the corresponding menu.

Fixes #5547.
Fixes #5641.

### Problem

1. The current page is not highlighted in the sidebar.
2. When opening a page from an expanded menu, the menu is collapsed again.

### Solution

1. Mark the current page with [aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current).
2. Open `<details>` elements that contain links to the current page.

---

## Screenshots

| Before | After |
|-------|-------|
| <img width="675" alt="image" src="https://user-images.githubusercontent.com/495429/188864171-a22a2521-69e1-4ac8-89da-ea43dda835dd.png"> | <img width="677" alt="image" src="https://user-images.githubusercontent.com/495429/189112863-c8eb0b71-395d-40c1-b3e1-5fecb19022cd.png">  |

---

## How did you test this change?

Opened these pages locally:

- http://localhost:3000/en-US/docs/Web/CSS/CSS_Animations
- http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Operators/Assignment
- http://localhost:3000/en-US/docs/Web/HTML/Global_attributes/class
- http://localhost:3000/en-US/docs/Web/HTTP/CSP
- http://localhost:3000/en-US/docs/Learn/CSS/Styling_text/Fundamentals
- http://localhost:3000/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks
- http://localhost:3000/en-US/docs/Learn/JavaScript/Building_blocks/conditionals

wbamberg commented:
This is great to see! cc @hamishwillee who had some good comments at https://github.com/mdn/yari/issues/5547#issuecomment-1067567913 .

It would be even better if we would scroll the sidebar so the highlighted link is visible. Is that planned as a follow-up?

A few suggestions.

1. I think this sort of thing works better when the highlight takes up the full width of the sidebar, so all highlights are the same width, whatever the length of the link title:

<img width="508" alt="Screen Shot 2022-09-07 at 4 08 04 PM" src="https://user-images.githubusercontent.com/432915/188998668-97c2b727-f0c9-4c4d-9cb9-38051c9d82b7.png">

2. It would be nice if the highlight in the sidebar used the same highlighting as the highlight-current-section in the ToC:

<img width="271" alt="Screen Shot 2022-09-07 at 4 09 23 PM" src="https://user-images.githubusercontent.com/432915/188998807-5d0c6d5f-14a5-4b01-bb6e-e13291d5a9c8.png">

As it is things look inconsistent.

3. The current highlighting has poor contrast: https://contrastchecker.com/?c=cdcdcd&b=696969.

caugner commented:
@wbamberg Good catch, and the highlighted headings didn't look good either with their padding, so I updated the strategy a bit, wrapping the links **and their siblings** in an `<em>` like the JSRef macro already used to do.

I have updated the screenshot in the PR description.

wbamberg commented:
I think this is much better, thank you! The border and left-padding indents the current page entry:

<img width="304" alt="Screen Shot 2022-09-08 at 4 30 40 PM" src="https://user-images.githubusercontent.com/432915/189243301-02b5873d-f33d-4232-b0f3-a362315b2972.png">

...this is a pretty nit-picky detail though so I won't insist on it :).

wbamberg reviewed (approved):
Approving but I'm happy for you to adjust things so the highlighted item is not indented, if you think it is worth doing. Thanks @caugner !

hamishwillee reviewed (approved):
@caugner Thank you. This is excellent - I've been complaining about how crappy our sidebar is for such a long time. I'd actually given up.

This is vastly better than the current situation. Tested:
- Opening a page by copying an a URL collapses the sidebar except for the section with the page, and highlights the page in sidebar. Excellent.
- Opening a page from sidebar does the same thing.
- - Opening a page from a link does the same thing.
- Sidebar highlight is clearly visible in all sections and easy to see if you scroll down the sidebar.
- Sidebar highlight good in both light and dark theme.

A few comments:

- Nothing is highlighted in sidebar for a top level entry point like http://localhost:3000/en-US/docs/Web/CSS . Should it be? Because I expected a sidebar entry I was looking.
- The different colour highlighting in different areas of the site do no harm, but add not benefit (IMO). I could be wrong for others.
- You can navigate the sidebar with a page open and the tree remain open until you select a new page. This is valid and fine. Note though that some people prefer that if you open one tree you close all others, even when "browsing" (I am one of those).
- The only flaw is that the sidebar doesn't scroll to visibility on opening - which you plan on addressing in a separate issue. 

I am approving this. Happy for it to be merged **right now**.

hamishwillee commented:
@wbamberg Thanks for pinging me in. FWIW Without scrolling to the current page I still don't think we can dump the Learn "QuickLinks". The current page is findable now, but you don't have the easy context: (from http://localhost:3000/en-US/docs/Learn/Server-side/Django/Authentication#in_this_module)

When we get to that point, I wonder if we should put the tutorial in its own "folder"? 
![image](https://user-images.githubusercontent.com/5368500/189251303-d305d1bf-0472-4a04-a0bc-e06c64784958.png)

caugner commented on build/document-extractor.ts:
```diff
@@ -36,14 +37,25 @@ type SectionsAndFlaws = [Section[], string[]];
   if (!search.length) {
-    return "";
+    doc.sidebarHTML = "";
   }
```
```suggestion
  }
  return;
```

caugner commented on build/document-extractor.ts:
```diff
@@ -36,14 +37,25 @@ type SectionsAndFlaws = [Section[], string[]];
+
   if (!search.length) {
-    return "";
+    doc.sidebarHTML = "";
```
```suggestion
    doc.sidebarHTML = "";
    return;
```
